custom-workbenches/example-build-custom-workbench.md

```diff
@@ -0,0 +1,303 @@
+[https://console.cloud.google.com/](https://console.cloud.google.com/)
+in a browser.
+
+Create a new GCP Project using the
+[Cloud Resource Manager](https://console.cloud.google.com/cloud-resource-manager).
+The project you create is just for this example, so you'll delete it below
+when you're done.
+
+You will need to
+[enable billing](https://support.google.com/cloud/answer/6293499#enable-billing)
+for this project.
+
+You need to enable Compute Engine and Vertex AI services as enabling these APIs
+allows you to create the required resources.
+
+[Enable Example Services](https://console.cloud.google.com/flows/enableapi?apiid=compute.googleapis.com,logging.googleapis.com,monitoring.googleapis.com,notebooks.googleapis.com,artifactregistry.googleapis.com,cloudbuild.googleapis.com,cloudresourcemanager.googleapis.com,containerregistry.googleapis.com,iam.googleapis.com,iap.googleapis.com)
+
+Note that some of these are needed in other examples but are included here so
+you can just use one project for all of these "custom workbench" examples.
+
+Next, make sure the project you just created is selected in the top of the
+Cloud Console.
+
+Then open a [Cloud Shell](https://cloud.google.com/shell) associated with the
+project you just created.
+
+[Launch CloudShell](https://console.cloud.google.com/?cloudshell=true)
+
+It's important that the current Cloud Shell project is the one you just
+created.  Verify that
+
+```bash
+echo $GOOGLE_CLOUD_PROJECT
+```
+
+shows that new project.
+
+All example commands below run from this Cloud Shell.
+
+
+## Example source
+
+The commands in this example are captured as scripts in a repo. You can
+cut/paste commands into your cloudshell instance or you can optionally just
+clone this repo into your cloudshell instance and run the scripts from there.
+
+Get the source
+
+```bash
+git clone https://github.com/GoogleCloudPlatform/scientific-computing-examples
+cd custom-workbenches
+```
+
+All example commands below are relative to this particular example directory.
+
+## Tools
+
+We use [gcloud](https://cloud.google.com/cli?hl=en) for these examples and the
+latest version is already installed in your GCP CloudShell.
+
+
+## Build a custom workbench image
+
+
+### Create an image repo to keep your custom images
+
+```bash
+REGION="us-central1"
+gcloud artifacts repositories create \
+  --location=${REGION} \
+  --repository-format="docker" \
+  "mycustomimagerepo"
+```
+
+The resulting registry URL should take the form:
+`<region>-<format>.pkg.dev/<project>/<repo_name>`.
+
+E.g.,
+```
+us-central1-docker.pkg.dev/mycoolprojectname/mycustomimagerepo
+```
+
+You can look it up using
+
+```bash
+gcloud artifacts repositories describe \
+  --location=${REGION} \
+  mycustomimagerepo
+```
+
+### Create a custom workbench image using docker
+
+Example `Dockerfile` to create a custom workbench image:
+
+```Dockerfile
+FROM gcr.io/deeplearning-platform-release/workbench-container:latest
+
+RUN pip install -q flask
+```
```

Comment:
I wasn't thinking this would be just click click click... I was intending they stop and copy this into a `Dockerfile`, same as below with the `cloudbuild.yaml` and edit both.

Reply:
So I provide these already as files in `images/mycustomimage/`, so I'm adding a note that points there.